### DISCUSSION TITLE
Only use traces in the dataset to render score columns

### DISCUSSION THREAD

**Discussion by micahstairs** [langfuse/langfuse#3415] [Ideas]
### Describe the feature or potential improvement

In the current implementation, datasets use the full history to render score columns. However, only the traces that are present in that dataset should be rendered in the score columns.

There are several reasons this could be useful:
1) If one or more score columns becomes obsolete. In my case, I used the API to publish scores in a few different columns as a test, but now I have no way to remove it from the UI, even after deleting all existing dataset runs.
2) If different score columns are relevant for different datasets. In my case, we intend on using different types of scoring for different datasets, but this is going to cause significant clutter in the UI, and possibly confuse users about what scores they should expect to see for a given dataset.

![Screenshot 2024-09-18 at 4 57 21 PM](https://github.com/user-attachments/assets/a85171a3-f991-4c4e-9dc9-8843b28269eb)

### Additional information

@marcklingen asked me to file a feature request for this in https://discord.com/channels/1111061815649124414/1286068813405880381/1286070441680703589

**Comment by marcklingen** [comment]
thank you!
